velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -228,4 +237,126 @@ FOLLY_ALWAYS_INLINE void PrefixSortEncoder::encodeNoNulls(
   encodeNoNulls(value.getNanos(), dest + 8);
 }
 
+class PrefixSortLongDecimalToIntEncoder : public PrefixSortEncoder {
+ public:
+  PrefixSortLongDecimalToIntEncoder(
+      bool ascending,
+      bool nullsFirst,
+      int precision,
+      int scale)
+      : PrefixSortEncoder(ascending, nullsFirst),
+        precision_(precision),
+        scale_(scale),
+        newScale_(newPrecision_ - (precision_ - scale_)) {
+    VELOX_DCHECK(precision - scale <= ShortDecimalType::kMaxPrecision);
+  };
+
+  FOLLY_ALWAYS_INLINE void encode(std::optional<int128_t> value, char* dest)
+      const override {
+    if (value.has_value()) {
+      dest[0] = nullsFirst_ ? 1 : 0;
+      int64_t newValue;
+      auto status = DecimalUtil::rescaleWithRoundUp<int128_t, int64_t>(
+          value.value(),
+          precision_,
+          scale_,
+          newPrecision_,
+          newScale_,
+          newValue);
+      if (FOLLY_UNLIKELY(!status.ok())) {
+        if (value < 0) {
+          newValue = std::numeric_limits<int64_t>::min();
+        } else {
+          newValue = std::numeric_limits<int64_t>::max();
+        }
+      }
+      encodeNoNulls(newValue, dest + 1);
+    } else {
+      dest[0] = nullsFirst_ ? 0 : 1;
+      simd::memset(dest + 1, 0, encodedSize() - 1);
+    }
+  }
+
+  static int32_t encodedSize() {
+    return 9;
+  }
+
+ private:
+  const int precision_;
+  const int scale_;
+  const int newPrecision_ = ShortDecimalType::kMaxPrecision;
+  const int newScale_;
+};
+
+class PrefixSortHugeIntEncoder : public PrefixSortEncoder {
+ public:
+  PrefixSortHugeIntEncoder(
+      bool ascending,
+      bool nullsFirst,
+      int precision,
+      int scale)
+      : PrefixSortEncoder(ascending, nullsFirst),
+        precision_(precision),
+        scale_(scale){};
+
+  FOLLY_ALWAYS_INLINE void encode(std::optional<int128_t> value, char* dest)
```

Comment:
encode() methhod is  "hot-code", please not add virtual call.

Could you try using this approach to handle different types of logic?
```
void RowContainer::store(
    const DecodedVector& decoded,
    vector_size_t index,
    char* row,
    int32_t column) {
  auto numKeys = keyTypes_.size();
  if (column < numKeys && !nullableKeys_) {
    VELOX_DYNAMIC_TYPE_DISPATCH(
        storeNoNulls,
        typeKinds_[column],
        decoded,
        index,
        row,
        offsets_[column]);
  } else {
    VELOX_DCHECK(column < keyTypes_.size() || accumulators_.empty());
    auto rowColumn = rowColumns_[column];
    VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
        storeWithNulls,
        typeKinds_[column],
        decoded,
        index,
        row,
        rowColumn.offset(),
        rowColumn.nullByte(),
        rowColumn.nullMask());
  }
}
```

Reply:
This virtual method is just for int128_t, other type is not virtual
https://github.com/facebookincubator/velox/pull/10385/files#diff-68588996b759bc7bd50b7eff21fcfc2b4ec7b67d234783b348a8af24cbd38a2fR59

Reply:
encoders is in SortLayout, so we cannot use non-virtual function for int128, we should change `extractRowColumnToPrefix` to `VELOX_DYNAMIC_TYPE_DISPATCH_ALL`

Reply:
Just an example, the "type dispatch code" is done at extractRowColumnToPrefix already, I think we just need to add    
'case branch' for new types.